force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,165 @@
+
+    public override void beforeDelete() {
+        //* Prevent deletion of closed Opportunities
+        //* When an opportunity is deleted prevent the deletion of a closed won opportunity if the account industry is 'Banking'.
+        Map<Id, Account> accounts = new Map<Id, Account>([SELECT Id, Industry FROM Account WHERE Id IN (SELECT AccountId FROM Opportunity WHERE Id IN :this.oldOppList)]);
+        for(Opportunity opp : this.oldOppList) {
+            if((opp.StageName == 'Closed Won' && accounts.get(opp.AccountId).Industry == 'Banking') || (opp.IsClosed)){
+                opp.addError('Cannot delete closed opportunity');
+            }
+        }
+    }
+
+    public override void afterDelete() {
+        //* Send email notifications when an Opportunity is deleted 
+        notifyOwnersOpportunityDeleted(this.oldOppList);
+    }
+
+    public override void afterUndelete() {
+        //* Assign the primary contact to undeleted Opportunities
+        assignPrimaryContact(this.newOppMap);
+    }
+
+
+    /*
+    notifyOwnersOpportunityDeleted:
+    - Sends an email notification to the owner of the Opportunity when it gets deleted.
+    - Uses Salesforce's Messaging.SingleEmailMessage to send the email.
+    */
+    private static void notifyOwnersOpportunityDeleted(List<Opportunity> opps) {
+        List<Messaging.SingleEmailMessage> mails = new List<Messaging.SingleEmailMessage>();
+        Map<Id, User> userMap = new Map<Id, User>([SELECT Id, Email FROM User]);
+        for (Opportunity opp : opps){
+            Messaging.SingleEmailMessage mail = new Messaging.SingleEmailMessage();            
+            //String[] toAddresses = new String[] {[SELECT Id, Email FROM User WHERE Id = :opp.OwnerId].Email};
+            User owner = userMap.get(opp.OwnerId);
+            String[] toAddresses = new List<String>{ owner.Email };
+            mail.setToAddresses(toAddresses);
+            mail.setSubject('Opportunity Deleted : ' + opp.Name);
+            mail.setPlainTextBody('Your Opportunity: ' + opp.Name +' has been deleted.');
+            mails.add(mail);
+        }        
+        
+        try {
+            Messaging.sendEmail(mails);
+        } catch (Exception e){
+            System.debug('Exception: ' + e.getMessage());
+        }
+    }
+
+    /*
+    assignPrimaryContact:
+    - Assigns a primary contact with the title of 'VP Sales' to undeleted Opportunities.
+    - Only updates the Opportunities that don't already have a primary contact.
+    */
+    private static void assignPrimaryContact(Map<Id,Opportunity> oppsMap) {
+        // Get all Account Ids from the opportunities
+        Set<Id> accountIds = new Set<Id>();
+        for (Opportunity opp : oppsMap.values()) {
+            accountIds.add(opp.AccountId);  
+        }
+
+        // Query contacts from accountIds with Title 'VP Sales'
+        List<Contact> primaryContacts = [SELECT Id, AccountId FROM Contact WHERE Title = 'VP Sales' AND AccountId = :accountIds];
```

Comment:
Is there a way to query account and opportunities together so you reduce the number of SOQLs